app/upgrades/commission/upgrades.go

```diff
@@ -0,0 +1,69 @@
+package commission
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	sk *stakingkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Starting upgrade for multi staking...")
+		fixMinCommisionRate(ctx, sk)
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+func fixMinCommisionRate(ctx sdk.Context, staking *stakingkeeper.Keeper) {
+	// Upgrade every validators min-commission rate
+	validators := staking.GetAllValidators(ctx)
+	minComm := sdk.MustNewDecFromStr(NewMinCommisionRate)
+	params := staking.GetParams(ctx)
+	params.MinCommissionRate = minComm
+
+	staking.SetParams(ctx, params)
+
+	for _, v := range validators {
+		//nolint
+		if v.Commission.Rate.LT(minComm) {
+			comm, err := updateValidatorCommission(ctx, staking, v, minComm)
+			if err != nil {
+				panic(err)
+			}
+
+			v.Commission = comm
```

Comment:
lets move this line down to after `staking.BeforeValidatorModified(ctx, v.GetOperator())` just to be safe

Reply:
Yes